OpenSim/Common/ComponentList.h

```diff
@@ -76,6 +78,21 @@ class ComponentFilterMatchAll : public ComponentFilter {
         return new ComponentFilterMatchAll(*this);
     }
 };
+
+/** A component is considered a match if its full path name matches the 
+provided regular expression. */
+class ComponentFilterFullPathNameRegex : public ComponentFilter {
+public:
+    ComponentFilterFullPathNameRegex(const std::regex& expr) : _expr(expr) {}
```

Comment:
How about if I create a second constructor that takes a string? I'd like to leave one constructor that takes a regex, as someone could conceivably want to [set their own flags](http://www.cplusplus.com/reference/regex/basic_regex/basic_regex/) on the regex (e.g., case-insensitive).